stage1/usr_from_kvm/lkvm/patches/dead_lock_fix.patch

```diff
@@ -0,0 +1,57 @@
+diff --git a/builtin-run.c b/builtin-run.c
+index 54e03be..40d9a17 100644
+--- a/builtin-run.c
++++ b/builtin-run.c
+@@ -633,18 +633,42 @@ static struct kvm *kvm_cmd_run_init(int argc, const char **argv)
+
+ static int kvm_cmd_run_work(struct kvm *kvm)
+ {
+-	int i;
+-	void *ret = NULL;
+-
+-	for (i = 0; i < kvm->nrcpus; i++) {
+-		if (pthread_create(&kvm->cpus[i]->thread, NULL, kvm_cpu_thread, kvm->cpus[i]) != 0)
+-			die("unable to create KVM VCPU thread");
+-	}
+-
+-	/* Only VCPU #0 is going to exit by itself when shutting down */
+-	return pthread_join(kvm->cpus[0]->thread, &ret);
++        int i;
++        int exit_status = 0;
++
++        for (i = 0; i < kvm->nrcpus; i++) {
++                if (pthread_create(&kvm->cpus[i]->thread, NULL, kvm_cpu_thread, kvm->cpus[i]) != 0)
++                        die("unable to create KVM VCPU thread");
++        }
++
++        /*
++         * Only VCPU #0 is going to exit by itself when shutting down
++         *
++         * BUT It's not actually true for reboot sequence
++         * Reboot is sending SIGKVMEXIT signal to all cpus
++         * and that is causing race condition on removing
++         * devices (which are pausing exiting cpus).
++         * This piece of code waits for death of all
++         * cpu threads
++         */
++        for (i = 0; i < kvm->nrcpus; i++) {
++                int c_thrstatus = 0, retval = 0;
++
++                void* thrstatus = &c_thrstatus;
++
++                retval = pthread_join(kvm->cpus[i]->thread, &thrstatus);
++
++                if (retval != 0)
++                        die("unable to end KVM VCPU thread");
++
++                /* Set exit status if one of vcpus returns error code > 0 */
++                if ((intptr_t) c_thrstatus != 0)
```

Comment:
@krnowak Thanks for your review!
1, 2 & 3 - sure, I will check that and publish new changes soon
4 - I know about that and this is my intention. If one of `pthread_join`s fails I'm not able to do anything with VM(VM is propably dead or its threads cannot be joined), so I want to immediately close whole lkvm process with `die()`. Because of that, we can expose worst exit status from cpu threads.